tests/bluetooth/tester/src/btp_micp.c

```diff
@@ -0,0 +1,534 @@
+#if defined(CONFIG_BT_MICP_MIC_CTLR_AICS)
+static struct bt_micp_included micp_included;
+
+#endif /* CONFIG_BT_MICP_MIC_CTLR_AICS */
+
+static void btp_send_micp_found_ev(struct bt_conn *conn, uint16_t mute_handle)
+{
+	struct btp_micp_discovered_ev ev;
+	struct bt_conn_info info;
+
+	(void) bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.mute_handle = mute_handle;
+
+	tester_event(BTP_SERVICE_ID_MICP, BTP_MICP_DISCOVERED_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_micp_aics_handle_ev(struct bt_conn *conn, uint16_t state_handle,
+					 uint16_t gain_handle, uint16_t type_handle,
+					 uint16_t status_handle, uint16_t control_handle,
+					 uint16_t desc_handle)
+{
+	struct btp_micp_aics_handle_ev ev;
+	struct bt_conn_info info;
+
+	(void) bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.state_handle = state_handle;
+	ev.gain_handle = gain_handle;
+	ev.type_handle = type_handle;
+	ev.status_handle = status_handle;
+	ev.control_handle = control_handle;
+	ev.desc_handle = desc_handle;
+
+	tester_event(BTP_SERVICE_ID_MICP, BTP_MICP_AICS_CHRC_EV, &ev, sizeof(ev));
+
+}
+
+static void btp_send_micp_mute_state_ev(struct bt_conn *conn, uint8_t mute)
+{
+	struct btp_micp_mute_read_ev ev;
+	struct bt_conn_info info;
+
+	(void) bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.mute = mute;
+
+	tester_event(BTP_SERVICE_ID_MICP, BTP_MICP_MUTE_READ_EV, &ev, sizeof(ev));
+
+}
+
+static void btp_send_aics_state_ev(struct bt_conn *conn, int8_t gain, uint8_t mute, uint8_t mode)
+{
+	struct btp_micp_aics_state_ev ev;
+	struct bt_conn_info info;
+
+	(void) bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.gain = gain;
+	ev.mute = mute;
+	ev.mode = mode;
+
+	tester_event(BTP_SERVICE_ID_MICP, BTP_MICP_AICS_STATE_EV, &ev, sizeof(ev));
+}
+
+static void micp_mic_ctlr_discover_cb(struct bt_micp_mic_ctlr *mic_ctlr, int err,
+				      uint8_t aics_count)
+{
+	struct bt_conn *conn;
+	uint16_t mute_handle;
+
+	if (err != 0) {
+		LOG_DBG("Discovery failed (%d)", err);
+	} else {
+		LOG_DBG("Discovery done with %u AICS",
+			aics_count);
+
+		bt_micp_mic_ctlr_conn_get(mic_ctlr, &conn);
+
+#if defined(CONFIG_BT_MICP_MIC_CTLR_AICS)
+		if (bt_micp_mic_ctlr_included_get(mic_ctlr, &micp_included) != 0) {
+			LOG_DBG("Could not get included services");
+			}
+
+		uint16_t state_handle;
+		uint16_t gain_handle;
+		uint16_t type_handle;
+		uint16_t status_handle;
+		uint16_t control_handle;
+		uint16_t desc_handle;
+
+		bt_micp_mic_ctlr_aics_chrc_get(micp_included, &state_handle, &gain_handle,
+					       &type_handle, &status_handle, &control_handle,
+					       &desc_handle);
+		btp_send_micp_aics_handle_ev(conn, state_handle, gain_handle, type_handle,
+					     status_handle, control_handle, desc_handle);
+#endif /* CONFIG_BT_MICP_MIC_CTLR_AICS */
+	}
+	bt_micp_mic_ctlr_chrc_get(mic_ctlr, &mute_handle);
```

Comment:
```suggestion
	}
	
	bt_micp_mic_ctlr_chrc_get(mic_ctlr, &mute_handle);
```

We usually keep an empty line after a `}` unless the line is also a `}`